src/player.js

```diff
@@ -3,38 +3,88 @@ class Character {
         this.controller = controller;
     }
 
     update() {
-        // Stop any previous movement from the last frame
-        this.sprite.body.setVelocity(0);
-        // Horizontal movement
-        if (this.controller.left.isDown) {
-            this.sprite.body.setVelocityX(-this._speed);
-            this.sprite.anims.play('playerLeft', true);
-        } else if (this.controller.right.isDown) {
-            this.sprite.body.setVelocityX(this._speed);
-            this.sprite.anims.play('playerRight', true);
-        } else if (this.controller.up.isDown) {
-            this.sprite.body.setVelocityY(-this._speed);
-            this.sprite.anims.play('playerUp', true);
-        } else if (this.controller.down.isDown) {
-            this.sprite.body.setVelocityY(this._speed);
-            this.sprite.anims.play('playerDown', true);
-        } else {
-            this.sprite.anims.stopOnRepeat();
+        this._handleMovement();
+    }
+
+    isBlocked() {
+        return this._blocked.down || this._blocked.up || this._blocked.left || this._blocked.right;
+    }
+
+    _moveWithStep(key, stepSizeX, stepSizeY) {
+        this._isMoving = true;
+        this._destination.x = this.sprite.x + stepSizeX;
+        this._destination.y = this.sprite.y + stepSizeY;
+        this.sprite.anims.play(key, true);
+    }
+
+    _hasArrived(error) {
+        return !this._isMoving ||
+            is_between(this._destination.x, this._prevPosition.x, this.sprite.x, error) &&
+            is_between(this._destination.y, this._prevPosition.y, this.sprite.y, error);
+    }
+
+    _handleMovement() {
+        const allowedError = 0.33;  // allowed error in pixels
+        const stepSize = 16;
+        if (this._hasArrived(allowedError) || this.isBlocked()) {
+            this._isMoving = false;
+        }
+        if (!this._isMoving) {
+            // Stop any previous movement from the last frame
+            this.sprite.body.setVelocity(0);
+            // Horizontal movement
+            if (this.controller.left.isDown) {
+                this.sprite.body.setVelocityX(-this._speed);
+                this._moveWithStep('playerLeft', -stepSize, 0);
+            } else if (this.controller.right.isDown) {
+                this.sprite.body.setVelocityX(this._speed);
+                this._moveWithStep('playerRight', stepSize, 0);
+            } else if (this.controller.up.isDown) {
+                this.sprite.body.setVelocityY(-this._speed);
+                this._moveWithStep('playerUp', 0, -stepSize);
+            } else if (this.controller.down.isDown) {
+                this.sprite.body.setVelocityY(this._speed);
+                this._moveWithStep('playerDown', 0, stepSize);
+            } else {
+                this.sprite.anims.stopOnRepeat();
+            }
         }
+        this._prevPosition.x = this.sprite.x;
+        this._prevPosition.y = this.sprite.y;
+    }
+}
+
+function is_between(expected, xval, yval, error=0) {
```

Comment:
I believe that we should use cammelCase here. Olso for the parameters.

Reply:
sure